### PR TITLE
Use cfg-if to create logical feature gate implementations

### PR DESCRIPTION
**Pull request by kinggoesgaming**
**I'm submitting a ...**
  - [ ] bug fix
  - [ ] feature enhancement
  - [ ] deprecation or removal
  - [x] refactor

# Description
`uuid` now adds `cfg-if` as a dependency.  With this, `#[cfg(..)]` implementations are easier to understand, as they can be handled in a more logical view because the macro allows us to `if` statements to make the checks more rustic at the source level

# Motivation
Understanding feature gate implementation becomes easier.

# Tests
The CI builds passing should be plenty.

# Related Issue(s)
#147

### CODE REVIEW

**Comment by kinggoesgaming** [comment]
question: should I put various feature gated constructor `new` functions behind cfg_if` macros aswell or should we wait until a bigger refactor comes along?

**Comment by kinggoesgaming** [comment]
@uuid-rs/uuid I think this is logical enough for this PR. Should be ready for review and merge

File: src/lib.rs [hunk 1]
```diff
@@ -110,39 +110,75 @@
...
+cfg_if! {
+    if #[cfg(all(feature = "slog", not(test)))] {
+        extern crate slog;
+    } else if #[cfg(all(feature = "slog", test))] {
+        #[macro_use]
+        extern crate slog;
+    }
+}
+
+cfg_if! {
```
Comment (KodrAus) [inline src/lib.rs]: Maybe we could reduce some duplicated nesting here by conditionally using `core as std`:

```rust
if #[cfg(not(feature = "std"))] {
    use core as std;
}

use std::fmt;
use std::str;

...
```

What do you think?

Comment (kinggoesgaming) [reply src/lib.rs]: This PR was intended to switch to `cfg_if`. Another PR should make the that part more readable later... which I think I will have ready soon

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by kinggoesgaming** [comment]
clippy was outdated, cache cleared

bors r+

**Comment by kinggoesgaming** [comment]
again! wrong cache cleared :(

bors r+

**Pull request merged**